xcell/cls/cov.py

```diff
@@ -429,39 +431,90 @@ def get_covariance(self):
             cla2b2 = self._get_cl_for_cov(self.clA2B2, self.clfid_A2B2,
                                           m_a2, m_b2)
 
-        if np.any(cla1b1) or np.any(cla1b2) or np.any(cla2b1) or \
-                np.any(cla2b2):
+        notnull = (np.any(cla1b1) or np.any(cla1b2) or
+                   np.any(cla2b1) or np.any(cla2b2))
+        s_a1, s_a2 = self.clA1A2.get_spins()
+        s_b1, s_b2 = self.clB1B2.get_spins()
+
+        size1 = self.clA1A2.get_ell_cl()[1].size
+        size2 = self.clB1B2.get_ell_cl()[1].size
+        cov_G = np.zeros((size1, size2))
+        cov_NG = np.zeros((size1, size2))
+        cov_nlm = np.zeros((size1, size2))
+        cov_mm = np.zeros((size1, size2))
+        if notnull:
             wa = self.clA1A2.get_workspace_cov()
             wb = self.clB1B2.get_workspace_cov()
             cw = self.get_covariance_workspace()
 
-            s_a1, s_a2 = self.clA1A2.get_spins()
-            s_b1, s_b2 = self.clB1B2.get_spins()
             if self.spin0 and (s_a1 + s_a2 + s_b1 + s_b2 != 0):
-                cov = self._get_covariance_spin0_approx(cw, s_a1, s_a2, s_b1,
-                                                        s_b2, cla1b1, cla1b2,
-                                                        cla2b1, cla2b2, wa,
-                                                        wb)
+                cov_G = self._get_covariance_spin0_approx(cw, s_a1, s_a2, s_b1,
+                                                          s_b2, cla1b1, cla1b2,
+                                                          cla2b1, cla2b2, wa,
+                                                          wb)
 
             else:
-                cov = nmt.gaussian_covariance(cw, s_a1, s_a2, s_b1, s_b2,
-                                              cla1b1, cla1b2, cla2b1, cla2b2,
-                                              wa, wb)
-        else:
-            size1 = self.clA1A2.get_ell_cl()[1].size
-            size2 = self.clB1B2.get_ell_cl()[1].size
-            cov = np.zeros((size1, size2))
+                cov_G = nmt.gaussian_covariance(cw, s_a1, s_a2, s_b1, s_b2,
+                                                cla1b1, cla1b2, cla2b1, cla2b2,
+                                                wa, wb)
+
+        if self.nl_marg and notnull:
+            cov_nlm = self.get_covariance_nl_marg()
 
-        if self.nl_marg:
-            cov += self.get_covariance_nl_marg()
+        if self.m_marg and notnull:
+            cov_mm = self.get_covariance_m_marg()
 
-        if self.m_marg:
-            cov += self.get_covariance_m_marg()
+        if self.do_NG and notnull:
+            fsky = self.data.data['cov'].get('fsky_NG', -1.)
```

Comment:
Not sure I like using numbers for these cases. I don't think the meaning will be obvious.  I'd prefer to use `None`.

Reply:
Done

Reply:
Thanks